docs/contributing/using-instrumenter-api.md

```diff
@@ -116,6 +116,16 @@ The `builder()` method accepts three arguments:
 An `Instrumenter` can be built from several smaller components. The following subsections describe
 all interfaces that can be used to customize an `Instrumenter`.
 
+### Set the instrumentation version and OpenTelemetry schema URL
+
+Always set the instrumentation library version on the `Instrumenter`. You can do this by calling
+the `setInstrumentationVersion()` method on the `InstrumenterBuilder`. By setting the version, you
+let users identify which version of your instrumentation produced the telemetry.
```

Comment:
do we want to promote the version-in-file as public best practice?

Reply:
I wasn't sure about that - I thought that it'd be the best to recommend the declarative `.setInstrumentationVersion()` way of setting it, while the properties file is sort of a hack that we use internally in all our instrumentations.
On the other hand, if we plan to make instrumentation-api stable it means that the properties file naming scheme (and the version property name) will also become stable - if we can't make backwards incompatible changes we might as well document it right away.
WDYT?

Reply:
I think it's a useful pattern to promote for protection against uberjars